### PR TITLE
feat: config schema generator

### PR DESCRIPTION
**Pull request by amrbashir**
### What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Docs
- [ ] New Binding issue #___
- [ ] Code style update
- [ ] Refactor
- [X] Build-related changes
- [ ] Other, please describe:

### Does this PR introduce a breaking change?

- [ ] Yes, and the changes were approved in issue #___
- [X] No

### Checklist
- [ ] When resolving issues, they are referenced in the PR's title (e.g `fix: remove a typo, closes #___, #___`)
- [ ] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/tauri/blob/dev/.changes/readme.md).
- [ ] I have added a convincing reason for adding this feature, if necessary

### CODE REVIEW

File: tooling/config-schema/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+name = "config-schema"
+version = "0.0.0"
+edition = "2021"
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+
+[build-dependencies]
+tauri-utils = { version = "1.0.0", features = [ "schema", "isolation" ], path = "../../core/tauri-utils" }
+schemars = { version = "0.8", features = [ "url", "preserve_order" ] }
+serde_json = "1.0"
```
Comment (amrbashir) [inline tooling/config-schema/Cargo.toml]:
```suggestion
tauri-utils = { version = "1.0.0", features = [ "schema", "isolation" ], path = "../../core/tauri-utils" }
schemars = { version = "0.8", features = [ "url", "preserve_order" ] }
serde = { version = "1.0", features = [ "derive" ] }
serde_json = "1.0"
serde_with = "2.0"
url = { version = "2.3", features = [ "serde" ] }
```

**Comment by amrbashir** [comment]
I am not sure why `test-tauri-core` fails with this PR when I made no changes to core

**Comment by lucasfernog** [comment]
You did change it though, I think it's messing with the tauri-utils features. I'll take a look.

**Comment by lucasfernog** [comment]
The biggest issue is adding the config-schema package to the workspace, which enables the `isolation` feature for the tauri-utils package even though it's not enabled for the other packages. I think we should find a different way to do this.

**Comment by amrbashir** [comment]
I could just remove it from the workspace, I mainly added it to the workspace so it can be autogenerated by the IDE's LSP when working with the config.

**Pull request merged**